internal/middleware/enforce_identity.go

```diff
@@ -65,19 +66,25 @@ func EnforceIdentity(next http.Handler) http.Handler {
 	})
 }
 
-func checkHeader(ctx context.Context, id *identity.XRHID, w http.ResponseWriter) error {
+func checkHeader(ctx context.Context, id *identity.XRHID, w http.ResponseWriter, r *http.Request) error {
 	if (id.Identity.Type == "Associate" || id.Identity.Type == "X509") && id.Identity.AccountNumber == "" {
 		return nil
 	}
 
 	if id.Identity.OrgID == "" && id.Identity.Internal.OrgID == "" {
-		doError(ctx, w, 400, "X-Rh-Identity header has an invalid or missing org_id")
-		return InvalidOrMissingOrgErr
+		errRender := render.Render(w, r, payloads.NewMissingIdentityError(r.Context(), "X-Rh-Identity header has an invalid or missing org_id", ErrIdentity))
+		if errRender != nil {
+			ctxval.Logger(ctx).Warn().Err(errRender).Msg("Cannot render permission middleware error")
+		}
+		return ErrIdentity
 	}
 
 	if id.Identity.Type == "" {
-		doError(ctx, w, 400, "X-Rh-Identity header is missing type")
-		return MissingTypeErr
+		errRender := render.Render(w, r, payloads.NewMissingIdentityError(r.Context(), "X-Rh-Identity header is missing type", ErrIdentity))
+		if errRender != nil {
+			ctxval.Logger(ctx).Warn().Err(errRender).Msg("Cannot render permission middleware error")
+		}
+		return ErrIdentity
 	}
 
 	return nil
```

Comment:
I am dropping the `doError` function, initially I thought I would reuse it in the other enforce middleware I am introducing, that did not work out, then I tried to use service function but import cycles. Then I decided to simplify it further - both middleware functions now use render package directly. It is slightly more verbose but I am not introducing middleware-service package dependency.